### ISSUE TITLE
Spotty right click handling on web

### ISSUE THREAD

**Issue by justinmc** [flutter/flutter#118741]
After right clicking a few times in a GestureDetector on web, I consistently get into a state where right clicks are no longer detected.

In the video below, a GestureDetector changes the color of the Container onSecondaryTapDown.  You can see it no longer works after a few times.

https://user-images.githubusercontent.com/389558/213299960-01ed417c-1c60-45fb-bd95-28303791571f.mov

This doesn't happen on desktop for me, only web.

### Steps to reproduce

  1. Run an app with a GestureDetector with an onSecondaryTap or onSecondaryTapDown handler (such as the one below).
  2. Right click the GestureDetector a few times.

Expect: The GestureDetector always detects right clicks.
Actual: Right clicks stop being detected after a handful of times.

<details>

<summary>Code</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        body: Center(
          child: _ClickMe(),
        ),
      ),
    );
  }
}

class _ClickMe extends StatefulWidget {
  @override
  State<_ClickMe> createState() => _ClickMeState();
}

class _ClickMeState extends State<_ClickMe> {
  bool clicked = true;

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onSecondaryTapDown: (TapDownDetails details) async {
        setState(() {
          clicked = !clicked;
        });
      },
      child: Container(
        width: 100,
        height: 100,
        color: clicked ? Colors.deepPurpleAccent : Colors.orangeAccent,
      ),
    );
  }
}

```

</details>

**Comment by yjbanov** [comment]
Is it possible that it happens exactly when the `pointerup` lands inside the pop-up menu? I believe @dkwingsmt had code to deal with this case.

**Comment by justinmc** [comment]
You're right!  I can reproduce it consistently if I right click down and then release it up while the pointer is inside of the menu.  I'll update the description above.

For the record I also tried this on Linux and I couldn't reproduce the bug.

**Comment by justinmc** [comment]
I talked to @dkwingsmt and he pointed me to [pointer_converter.dart](https://github.com/flutter/engine/blob/42eec6280719394db1b30bf3284b3fc7780588cb/lib/web_ui/lib/src/engine/pointer_converter.dart), which is likely getting confused when it sees a right mouse down with no up.
